### ISSUE TITLE
 #169 again ?

### ISSUE THREAD

**Issue by Coeur-Noir** [syncthing/syncthing-gtk#224]
Hello !

Did you change anything about emblems in v0.8 ?

On my 2 ubuntu 14.04 unity with nautilus integration,

emblems are only show on currently synching files (double blue circle arrows) and are not updated (they always look as they *are* synching).

No other emblems are ever shown, what ever directory, even after a nautilus -q

I have that same behavior on the 2 ubuntu 14.04 running nautilus (the third machine is Lubuntu with PcManFm).

![capture du 2015-11-24 12 53 29](https://cloud.githubusercontent.com/assets/1895947/11366011/c0c178f4-92aa-11e5-9bea-b182391efc6e.jpg)

**Comment by kozec** [comment]
Not really, nothing emblem related was changed as far as I'm aware. Try to get Nautilus output, maybe it's caching problem again.

**Comment by Coeur-Noir** [comment]
![capture du 2015-11-26 15 54 37](https://cloud.githubusercontent.com/assets/1895947/11425597/4bd80820-9456-11e5-9f4b-9f31f33217a2.jpg)

Not very interesting, isn't it ?

**Comment by kozec** [comment]
@Coeur-Noir I'm trying to replicate error on two machines and simply can't (as usual :D)

Weirdest thing is that you are not getting even errors you *should* get. Are you getting something like

```
E Daemon        g-io-error-quark: Could not connect to 127.0.0.1: Connection refused (39)
```

at least when Syncthing is turned off?

**Comment by Coeur-Noir** [comment]
/usr/local/lib/python2.7/site-packages → empty folder ?

nautilusplugin.py is to be found inside → /usr/lib/python2.7/dist-packages/syncthing_gtk

**Comment by kozec** [comment]
> nautilusplugin.py is to be found inside → /usr/lib/python2.7/dist-packages/syncthing_gtk

That should be it, location depends on distro and/or packager.

**Comment by Coeur-Noir** [comment]
Oh ! Emblems !

Good point !

**Comment by Coeur-Noir** [comment]
Have a question though. But I think the answer is related to syncthing itself.
![capture du 2015-11-29 13 02 00](https://cloud.githubusercontent.com/assets/1895947/11456915/a4f5342e-9699-11e5-9180-f723fdd7e5b7.jpg)
The highlighted "John Williams" folder on the picture, is the latest created here. It contains a dozen files about 135 Mo. I assume theses folder and files are the ones currently synching *with* POSEO-bureau-gerald.

So is the emblem correct ? Should not be it the "double-circle-blue-arrow" to mean there's something currently happening on these folder and files ?

**Comment by kozec** [comment]
Not really, Syncthing doesn't have "something is being sent" state. 
<img src="https://raw.githubusercontent.com/syncthing/syncthing-gtk/master/icons/emblem-syncthing-active.png">
... is displayed only on file that is being downloaded and on repository itself.

**Comment by Coeur-Noir** [comment]
Ok !

**Comment by Coeur-Noir** [comment]
So the ![emblem-syncthing-active](https://cloud.githubusercontent.com/assets/1895947/11499081/d4fb294a-9823-11e5-817b-37a8b17a0bee.png) once updating/downloading done should turn back into the full blue circle emblem, shouldn't it ?

(Doing a nautilus -q and restart nautilus → full blue circle emblems)

I remind also grey emblems that were displayed on files formerly sync'ed when syncthing itself is not yet connected to other devices. Gone ?

**Comment by kozec** [comment]
Grey emblem is by design. It's displayed on repo that are set to be shared, but all devices that is repo shared with are offline. Such repo is displayed with grey "Offline" state in ST-GTK as well.

Turning into blue circle should happen, but it depends also on nautilus re-requesting file state after it is invalidated.

**Comment by Coeur-Noir** [comment]
Turning into blue circle should happen, but it depends also on nautilus re-requesting file state after it is invalidated. → I've thought it was kind of fixed recently.

Thanx for your explanation and time ;-)

**Comment by Coeur-Noir** [comment]
So as far as I can see, Nautilus does not update emblems reliably : 

- if I stop Syncthing on the distant devices, emblems on local pc do immediatly turn grey, which is correct
- but files and folders with double-blue-arrow keep that emblem on,
- and anyway that double-blue-arrow should have previously turned back into a full-blue-emblem (since sync' is 100% on all devices).

So it seems the blue-double-arrow tends to stuck ?

**Comment by Coeur-Noir** [comment]
Is it related ? today terminal's output for nautilus tells : 

    palace@Asgard:~$ nautilus &
    [1] 4396
    palace@Asgard:~$ I SyncthingPlugin Initializing...
    I SyncthingPlugin Initialized.
    Initializing nautilus-open-terminal extension
    I SyncthingPlugin Connected to Syncthing daemon
    
    (nautilus:4396): GdkPixbuf-CRITICAL **: gdk_pixbuf_composite: assertion 'dest_y >= 0 && dest_y + dest_height <= dest->height' failed
    
    (nautilus:4396): GdkPixbuf-CRITICAL **: gdk_pixbuf_composite: assertion 'dest_y >= 0 && dest_y + dest_height <= dest->height' failed
    
    (nautilus:4396): GdkPixbuf-CRITICAL **: gdk_pixbuf_composite: assertion 'dest_y >= 0 && dest_y + dest_height <= dest->height' failed

**Comment by kozec** [comment]
I'm still trying to find way how to tell nautilus I really _do_ need to have those emblems refreshed, but it looks little moody :( Funny thing is that Nemo (Nautilus fork) doesn't have this problem.

Anyway, no, I don't think those messages are related to plugin.

**Comment by kozec** [comment]
Should be fixed by 638423948f8d18908fe8437d98d5f0df52cea7d1 and 623152e04c4dfbd06c783aeb0e98771974e17598.

**Comment by kozec** [comment]
... and fixed in [new release](https://github.com/syncthing/syncthing-gtk/releases/tag/v0.8.1) as well
